src/kernelfuture.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Jupyter Development Team.
```

Comment:
Why did you move this out of `kernel.ts` instead of just exporting the future implementation?